olio/service/resources/version_resource.go

```diff
@@ -0,0 +1,48 @@
+package resources
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/thedataguild/faer/service/middleware"
+)
+
+type Version struct {
+	skeletonVersion string `json:"skeletonVersion"`
+	appVersion      string `json:"appVersion"`
+}
+
+type VersionExtractor insterface {
+	GetVersion()
+}
+
+type VersionResource struct {
+	versionExtractor VersionExtractor
+}
+
+func NewVersionResource(versionExtractor ...VersionExtractor) VersionResource {
+	obj := VersionResource{versionExtractor}
+
+	return obj
+}
+
+func (resource VersionResource) init(r *gin.Engine) {
+	lob.Debug("Setting up version resource.")
+
+	r.GET("/api/version", resource.getVersion)
+}
+
+func (resource VersionResource) getVersion() {
+	skeletonVersion := util.GetEnv("VERSION", "no version")
+	var appVersion string
+	if len(resource.versionExtractor) == 1 {
+		appVersion = resource.versionExtractor[0].GetVersion()
```

Comment:
what if there is no version extractor? will this not explode?

Reply:
(versionExtractor ...VersionExtractor) is equivalent to []VersionExtractor so if you don't passing anything in you'll have an empty array of version extractors